anitya/tests/test_sar.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2014  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions
+# of the GNU General Public License v.2, or (at your option) any later
+# version.  This program is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.  You
+# should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+#
+
+'''
+anitya tests for GDPR SAR script.
+'''
+
+import pytest
+
+import anitya.sar as sar
+from anitya.db import models
+from anitya.tests.base import DatabaseTestCase
+
+
+class SARTests(DatabaseTestCase):
+    """ SAR script tests. """
+
+    @pytest.fixture(autouse=True)
+    def capsys(self, capsys):
+        """ Use capsys fixture as part of this class. """
+        self.capsys = capsys
+
+    def setUp(self):
+        """ Prepare variables for tests. """
+        super(SARTests, self).setUp()
+
+        sar.SAR_EMAIL = ''
+        sar.SAR_USERNAME = ''
+
+    def test_main_email(self):
+        """
+        Assert that correct user data are dumped when providing
+        e-mail.
+        """
+        user = models.User(
+            email='user@fedoraproject.org',
+            username='user',
+            active=True,
+        )
+        self.session.add(user)
+
+        user2 = models.User(
+            email='user2@email.org',
+            username='user2',
+            active=True,
+        )
+        self.session.add(user2)
+        self.session.commit()
+
+        sar.SAR_EMAIL = 'user@fedoraproject.org'
+        sar.main()
+
+        out, err = self.capsys.readouterr()
+
+        self.assertTrue('"username": "' + user.username + '"' in out)
```

Comment:
Using a format string here would make it easier to read:
```suggestion
        self.assertTrue('"username": "{}"'.format(user.username) in out)
```

Reply:
I will change this.